packages/enzyme-adapter-react-15.4/src/ReactFifteenFourAdapter.js

```diff
@@ -252,6 +253,15 @@ class ReactFifteenFourAdapter extends EnzymeAdapter {
   createElement(...args) {
     return React.createElement(...args);
   }
+
+  invokeSetStateCallback(instance, callback) {
+    // React in >= 15.4, and < 16 pass undefined to a setState callback
+    ifReact(
+      '^15.4',
+      () => { callback.call(instance, undefined); },
+      () => { super.invokeSetStateCallback(instance, callback); },
+    );
```

Comment:
`ifReact` returns a function, not calling it so you have to call a function which is returned from `ifReact`.

https://github.com/airbnb/enzyme/blob/master/packages/enzyme-adapter-react-helper/src/ifReact.js#L7

Reply:
lol, oops. shouldn't tests have failed?

Reply:
Will fix in a followup.

Reply:
I think so we should add a test to guarantee to call a setState callback.
Currently, we don't have a test for it.